packages/pg/lib/stream.js

```diff
@@ -0,0 +1,176 @@
+var { EventEmitter } = require('events')
+
+/**
+ * Get a socket stream compatible with the current runtime environment.
+ * @returns {Duplex}
+ */
+module.exports.getStream = function getStream(allowSsl) {
+  try {
+    var net = require('net')
+    if (typeof net.Socket === 'function') {
+      return new net.Socket()
+    }
+  } catch (e) {
+    // `net` module is not available
+  }
+  return new CFSocket(allowSsl)
+}
+
+module.exports.getSecureStream = function getSecureStream(options) {
+  var tls = require('tls')
+  if (tls.connect) {
+    return tls.connect(options)
+  } else {
+    options.socket.startTls(options)
+    return options.socket
+  }
+}
+
+/**
+ * Wrapper around the Cloudflare built-in socket that can be used by the `Connection`.
```

Comment:
Looks like something that could go in another package.

Reply:
Moved to a new pg-cloudflare package